xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateVersioningStore.java

```diff
@@ -351,11 +372,39 @@ public XWikiRCSNodeContent loadRCSNodeContent(final XWikiRCSNodeId id, boolean b
     public void deleteArchive(final XWikiDocument doc, boolean bTransaction, XWikiContext context) throws XWikiException
     {
         executeWrite(context, session -> {
-            session
-                .createQuery("delete from " + XWikiRCSNodeInfo.class.getName() + " where id." + FIELD_DOCID + '=' + ':'
-                    + FIELD_DOCID)
-                .setParameter(FIELD_DOCID, doc.getId()).executeUpdate();
+            VersioningStoreQueryFactory.getDeleteArchiveQuery(session, doc.getId()).executeUpdate();
             return null;
         });
     }
+
+    @Override
+    public long getXWikiDocVersionsCount(XWikiDocument doc, RevisionCriteria criteria, XWikiContext context)
+        throws XWikiException
+    {
+        return getRCSNodeInfoCount(context, doc.getId(), criteria);
+    }
+
+    /**
+     * Counts the number of RCS nodes present in the archive of a given document, based on criteria.
+     *
+     * @param context the XWiki context
+     * @param id {@link XWikiRCSNodeContent#getId()}
+     * @param criteria the criteria matching the nodes to count
+     * @return the number of matching RCS nodes
+     * @throws XWikiException if any error
+     */
+    protected long getRCSNodeInfoCount(XWikiContext context, final long id, RevisionCriteria criteria)
+        throws XWikiException
+    {
+        return executeRead(context, session -> {
+            try {
+                return VersioningStoreQueryFactory.getRCSNodeInfoCountQuery(session, id, criteria).getSingleResult();
+            } catch (IllegalArgumentException e) {
+                // This happens when the database has wrong values...
+                LOGGER.error("Encountered invalid history when computing archive size for document [{}]", id, e);
```

Comment:
I'm not sure if we want to capture and log this exception. Shouldn't we re-throw it in a `XWikiException`?

Reply:
The comment is more cryptic than the error message below it! If it doesn't bring anything more then it should be removed IMO

Reply:
I agree, but I was concerned about https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateVersioningStore.java#L320 (which is why I copied the comment). Now, this comment has been introduced in https://github.com/xwiki/xwiki-platform/commit/725f4594285667b7dc777a874156d43b5458f1b8 and the code has been refactored a lot since then so it might no longer be relevant (but I can't be sure).